bin/build-fonts.js

```diff
@@ -0,0 +1,289 @@
+    service: 'system',
+  },
+  {
+    family: 'Palatino',
+    fallbacks: [
+      'Palatino Linotype',
+      'Palatino LT STD',
+      'Book Antiqua',
+      'Georgia',
+      'serif',
+    ],
+    weights: DEFAULT_WEIGHTS,
+    styles: DEFAULT_STYLES,
+    service: 'system',
+  },
+  {
+    family: 'Papyrus',
+    fallbacks: ['fantasy'],
+    weights: DEFAULT_WEIGHTS,
+    styles: DEFAULT_STYLES,
+    service: 'system',
+  },
+  {
+    family: 'Tahoma',
+    fallbacks: ['Verdana', 'Segoe', 'sans-serif'],
+    weights: DEFAULT_WEIGHTS,
+    styles: DEFAULT_STYLES,
+    service: 'system',
+  },
+  {
+    family: 'Times New Roman',
+    fallbacks: ['Times New Roman', 'Times', 'Baskerville', 'Georgia', 'serif'],
+    weights: DEFAULT_WEIGHTS,
+    styles: DEFAULT_STYLES,
+    service: 'system',
+  },
+  {
+    family: 'Trebuchet MS',
+    fallbacks: [
+      'Lucida Grande',
+      'Lucida Sans Unicode',
+      'Lucida Sans',
+      'Tahoma',
+      'sans-serif',
+    ],
+    weights: DEFAULT_WEIGHTS,
+    styles: DEFAULT_STYLES,
+    service: 'system',
+  },
+  {
+    family: 'Verdana',
+    fallbacks: ['Geneva', 'sans-serif'],
+    weights: DEFAULT_WEIGHTS,
+    styles: DEFAULT_STYLES,
+    service: 'system',
+  },
+];
+
+const fonts = [...SYSTEM_FONTS];
+
+const FONT_VARIANT_PATTERN = /(?<weight>\d+)?(?<style>\D+)?/;
+
+/**
+ * Returns a valid CSS font fallback declaration for a Google font.
+ *
+ * @param {string} category Font category.
+ * @return {string} Font fallback.
+ */
+function getFontFallback(category) {
+  switch (category) {
+    case 'sans-serif':
+      return 'sans-serif';
+    case 'handwriting':
+    case 'display':
+      return 'cursive';
+    case 'monospace':
+      return 'monospace';
+    default:
+      return 'serif';
+  }
+}
+
+for (const font of googleFonts) {
+  const variants = [];
+  const weights = [];
+  const styles = [];
+
+  const { family, category, variants: initialVariants } = font;
+
+  for (const variant of initialVariants) {
+    const found = variant.match(FONT_VARIANT_PATTERN);
+
+    const weight = found.groups.weight || false;
+
+    if (weight) {
+      weights.push(Number(weight));
+    }
+
+    const style = found.groups.style || false;
+
+    if (style) {
+      if ('regular' === style || !weight) {
+        weights.push(400);
+      }
+
+      styles.push(style);
+    }
+
+    // Example: [ 1, 700] for italic+bold.
+    const variantTuple = [
+      Number('italic' === style),
+      weight ? Number(weight) : 400,
+    ];
+
+    const hasVariant = variants.some(
+      (val) => val[0] === variantTuple[0] && val[1] === variantTuple[1]
+    );
+
+    if (!hasVariant) {
+      variants.push(variantTuple);
+    }
+  }
+
+  fonts.push({
+    family,
+    fallbacks: [getFontFallback(category)],
+    weights: [...new Set(weights)],
+    styles: [...new Set(styles)],
+    variants,
+    service: 'fonts.google.com',
+  });
+}
+
+fonts.sort((a, b) => {
+  if (a.family > b.family) {
```

Comment:
This should probably do a case-insensitive ordering? Or maybe even better `localeCompare`?

Reply:
`localeCompare` would need to be done on the client-side too though for 100% correct results